sdk/appconfiguration/Azure.ResourceManager.AppConfiguration/tests/Tests/PrivateEndpointConnectionCollectionTests.cs

```diff
@@ -54,7 +52,22 @@ public async Task TestSetUp()
                 vnetData.AddressPrefixes.Add("10.0.0.0/16");
                 vnetData.DhcpOptionsDnsServers.Add("10.1.1.1");
                 vnetData.DhcpOptionsDnsServers.Add("10.1.2.4");
-                VirtualNetworkResource vnet = (await ResGroup.GetVirtualNetworks().CreateOrUpdateAsync(WaitUntil.Completed, VnetName, vnetData)).Value;
+                //VirtualNetworkResource vnet = (await ResGroup.GetVirtualNetworks().CreateOrUpdateAsync(WaitUntil.Completed, VnetName, vnetData)).Value;
+                ResourceIdentifier subnetID;
+                if (Mode == RecordedTestMode.Playback)
+                {
+                    subnetID = SubnetResource.CreateResourceIdentifier(ResGroup.Id.SubscriptionId, ResGroup.Id.Name, VnetName, SubnetName);
+                }
+                else
+                {
+                    using (Recording.DisableRecording())
+                    {
+                        var vnetResource = await ResGroup.GetVirtualNetworks().CreateOrUpdateAsync(WaitUntil.Completed, VnetName, vnetData);
+                        var subnetCollection = vnetResource.Value.GetSubnets();
+                        //SubnetResource subnetResource = (await subnetCollection.CreateOrUpdateAsync(WaitUntil.Completed, subnetName2, subnetData)).Value;
+                        subnetID = vnetResource.Value.Data.Subnets[0].Id;
+                    }
+                }
```

Comment:
Could we extract this part into a method and put it into the test base class (should be `AppConfigurationClientBase`) in this RP? We could make it `protected` so that we could use it in any derived class.
This method could look like this:
```csharp
protected Task<ResourceIdentifier> CreateVirtualNetwrokToGetSubnetId(ResourceIdentifier resourceGroupId, string vnetName, string subnetName, VirtualNetworkData vnetData)
```

Reply:
fixed